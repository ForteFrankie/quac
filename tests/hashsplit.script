#!/bin/bash

# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR

set +e
exec 2>&1  # send stderr to stdout to make output more readable


# Check argument parsing

x hashsplit
echo $?
x hashsplit 2
echo $?
x hashsplit foo
echo $?
x hashsplit 0 foo
echo $?
y "hashsplit '' foo"
echo $?
y "hashsplit 2 ''"
echo $?


# Check failure on can't open output file(s)
y "hashsplit 2 /does/not/exist < /dev/null"
echo $?

# Check that it works (note that we repeat "a" several times to make sure
# we're hashing just the key and not the whole line)

cat > in.txt <<EOF
a	1
b	2
c	3
a	4
b	5
a	6
a	7
a	8
a	9
a	10
a	11
a	12
a	13
a	14
a	15
a	16
a	17
a	18
a	19
a	20
a	21
nullvaluewithtab	
nullvaluenotab
EOF

x cat in.txt
y "hashsplit 2 out < in.txt"
x cat out.0
x cat out.1


# exit with success
exit 0
