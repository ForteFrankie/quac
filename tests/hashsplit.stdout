$ cat in.txt
a	1
b	2
c	3
a	4
b	5
nullvaluewithtab	
nullvaluenotab
$ (hashsplit 2 out < in.txt)
$ cat out.0
a	1
c	3
a	4
$ cat out.1
b	2
b	5
nullvaluewithtab	
nullvaluenotab
